### PR TITLE
Update deps and add new helpers

### PR DESCRIPTION
**Pull request by job13er**
### This project uses [semver](semver.org), please check the scope of this pr:
 - [ ] #patch# - backwards-compatible bug fix
 - [x] #minor# - adding functionality in a backwards-compatible manner
 - [ ] #major# - incompatible API change

# CHANGELOG
* **Updated** - `ember-mocha` dependency
* **Added** - Helpers to make using `setupTest()`, `setupComponentTest()` and `setupModelTest()` easier.
* **Deprecated** - the `describeComponent()`, `describeModel()` and `describeModule()` helpers
* **Deprecated** - the `diffObject()` method (we'll hopefully incorporate it into a chai addon soon, see #12)

### CODE REVIEW

File: ember-cli-build.js [hunk 1]
```diff
@@ -17,12 +17,14 @@ module.exports = function (defaults) {
     }
   })
 
-  /*
-    This build file specifes the options for the dummy test app of this
-    addon, located in `/tests/dummy`
-    This build file does *not* influence how the addon or the app using it
-    behave. You most likely want to be modifying `./index.js` or app's build file
-  */
+  if (app.env === 'test') {
```
Comment (sandersky) [inline ember-cli-build.js:20]: Do we want to include blueprints that automatically adds these two bower dependencies to consumers and wires up their `ember-cli-build.js` to include this?

Comment (job13er) [reply ember-cli-build.js:20]: I'm not sure that's necessary here. If we *do* want that, I think it belongs in https://github.com/ciena-frost/ember-frost-test

Comment (sandersky) [reply ember-cli-build.js:20]: If so you'll probably find https://github.com/ciena-frost/ember-frost-bunsen/blob/master/blueprints/ember-frost-bunsen/index.js#L26-L34 useful.

Comment (job13er) [reply ember-cli-build.js:20]: Mainly b/c we don't rely on those for what this provides, I only used them to test the code provided here.

**Comment by job13er** [comment]
Fixes #11

**Comment by sandersky** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/ciena-blueplanet/ember-test-utils/pull-request/13/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-blueplanet/ember-test-utils)

**Pull request merged**